### PR TITLE
Some cleanup before bringing in edit

### PR DESCRIPTION
**Pull request by zarsky-broad**
To make #1828 a little smaller/simpler.

1. A bit of cleanup around data table display & formatting.
2. ~`NumberInput` takes in and puts out numbers (or `null`) instead of strings.~
3. Cleanup & group entity-related ajax calls.
4. Simplify `Utils.convertValue` & make it more tolerant

### CODE REVIEW

File: src/components/input.js [hunk 1]
```diff
@@ -132,14 +132,15 @@ export const SearchInput = ({ value, onChange, ...props }) => {
...
-export const NumberInput = ({ onChange, ...props }) => {
+export const NumberInput = ({ onChange, value, ...props }) => {
   Utils.useConsoleAssert(props.id || props['aria-label'], 'In order to be accessible, NumberInput needs a label')
 
   return h(Interactive, _.merge({
     as: 'input',
     type: 'number',
     className: 'focus-style',
-    onChange: onChange ? (e => onChange(e.target.value)) : undefined,
+    onChange: onChange ? (e => onChange(_.toNumber(e.target.value))) : undefined,
```
Comment (panentheos) [inline src/components/input.js]: Calling `_.toNumber` unconditionally will convert `''` to `0`, which appears if you delete all the characters in the input. If we're going to make this change, then we probably want to return `null` or something in that case, and then handle that downstream.

File: src/components/input.js [hunk 2]
```diff
@@ -152,43 +153,23 @@ export const NumberInput = ({ onChange, ...props }) => {
...
+  }, [externalValue, needsUpdate])
 
-  static getDerivedStateFromProps({ value }, { lastValueProp }) {
-    if (value !== lastValueProp) {
-      return { textValue: value.toString(), lastValueProp: value }
+  return h(NumberInput, {
+    ...props, min, max, value: internalValue,
+    onChange: setInternalValue,
+    onBlur: () => {
+      setNeedsUpdate(true)
```
Comment (panentheos) [inline src/components/input.js]: Do we need this indirection, or could we just call `setInternalValue` here?

File: src/libs/utils.js [hunk 3]
```diff
@@ -258,11 +265,13 @@ export const cantBeNumber = _.flow(_.toNumber, _.isNaN)
...
     case 'string':
-      return value.toString()
+      // known issue where toString is incorrectly flagged:
+      // eslint-disable-next-line lodash-fp/preferred-alias
+      return _.toString(value)
     case 'number':
       return _.toNumber(value)
     case 'boolean':
-      return !(_.lowerCase(value) === 'false' || _.lowerCase(value) === 'no' || value === '0')
+      return !['false', 'no', '0', ''].includes(_.lowerCase(value))
```
Comment (ehanna4) [inline src/libs/utils.js:274]: TOL(?): It looks like array.protoype.includes() is not supported in IE, it seems like using array.protoype.indexof() which is supported by IE would work in this case. However, it is not as clear/quick to comprehend [].indexof(value) === -1 as it is to read ![].includes(value) so if supporting IE is not absolutely necessary then this likely wouldn't be a worthwhile swap though

Comment (zarsky-broad) [reply src/libs/utils.js:274]: Yup, we explicitly don't support it.

Comment (breilly2) [reply src/libs/utils.js:274]: Question: Are we talking IE or Edge here?

TOL: "Explicitly don't support" or "don't explicitly support"? I think there's a huge difference. For example, I think we "don't explicitly support" Firefox but we still do our best. Can't we have the same stance for all modern browsers?

Comment (ehanna4) [reply src/libs/utils.js:274]: @breilly2 I was speaking about Internet Explorer not supporting includes(), however I appreciate your second point and I would also like to know in general what the team stance is on cross-browser compatibility and what we are aiming to support

Comment (zarsky-broad) [reply src/libs/utils.js:274]: (discussed in person)

File: src/components/input.js [hunk 4]
```diff
@@ -152,43 +153,22 @@ export const NumberInput = ({ onChange, ...props }) => {
...
-  static getDerivedStateFromProps({ value }, { lastValueProp }) {
-    if (value !== lastValueProp) {
-      return { textValue: value.toString(), lastValueProp: value }
+  return h(NumberInput, {
+    ...props, min, max, value: internalValue,
+    onChange: setInternalValue,
+    onBlur: () => {
+      const newVal = _.clamp(min, max, _.floor(internalValue))
+      setInternalValue(newVal)
+      onChange(newVal)
```
Comment (petesantos) [inline src/components/input.js]: TOL - should we rename this to an "onBlur" handler so it is clearer in the consuming code when this handler will be executed? I wouldn't want to see it in this PR, I'd prefer to keep the diff down, but maybe in a follow up.

**Review by petesantos — APPROVED** [review]

**Pull request merged**